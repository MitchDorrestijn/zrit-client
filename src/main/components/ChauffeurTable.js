/**
 * React related imports
 */
import React from 'react';
import {Button} from 'reactstrap'
import {CSVLink} from 'react-csv';
import {BootstrapTable, TableHeaderColumn} from 'react-bootstrap-table';
import {renderSearchField, renderSortedColumn} from '../global/Methods';

/**
 * Endpoints import
 */
import {
  getAllChauffeurs
} from '../CRUD/Chauffeur';

/**
 * Style related imports
 */
import '../../../node_modules/react-bootstrap-table/dist/react-bootstrap-table-all.min.css';
import '../css/table.css';

/**
 * This class takes care of rendering the chauffeur table, from here other actions can be taken
 */
export default class ChauffeurTable extends React.Component {

  /**
   * Makes a GET request to get all chauffeurs when component is mounted
   */
  componentDidMount() {
    getAllChauffeurs(this.props).then((res) => {res !== undefined && this.setState({data: res.data})});
  }

  /**
   * Setups all the data.
   * @constructor
   * @param {props} props - The given properties by the superclass
   */
  constructor(props) {
    super(props);
    this.columns = [
      {
        name: 'name',
        display: 'Naam chauffeur'
      }, {
        name: 'totalRides',
        display: 'Aantal ritten'
      }, {
        name: 'typeOfPayment',
        display: 'Soort vergoeding'
      }, {
        name: 'totalEarned',
        display: 'Totaal verdiend'
      }, {
        name: 'numberPlate',
        display: 'Kenteken auto'
      }, {
        name: 'numberOfPassengers',
        display: 'Aantal personen auto'
      }, {
        name: 'segment',
        display: 'Segment auto'
      }, {
        name: 'rating',
        display: 'Beoordeling'
      }
    ];
    this.state = {
      sortColumnName: undefined,
      sortOrder: undefined,
      disableButtons: true,
      data: []
    };
  }

  /**
   * Renders the buttons above the table
   */
  renderButtons = () => {
    return (<div>
      <Button color="primary" className='crud-btn'>Toevoegen</Button>
      <Button disabled={this.state.disableButtons} color="primary" className='crud-btn'>Bewerken</Button>
    </div>);
  };

  /**
   * Renders the view for the user
   */
  render() {
    const tableOptions = {
      sortColumnName: this.state.sortName,
      sortOrder: this.state.sortOrder,
      renderSortedColumn: renderSortedColumn,
      noDataText: 'Geen resultaten gevonden',
      searchField: renderSearchField,
      btnGroup: this.renderButtons
    };

    return (<div>
      <BootstrapTable search={true} data={this.state.data} options={tableOptions} selectRow={{
          mode: 'radio',
          onSelect: this.onSelect
        }} ref='table'>

        <TableHeaderColumn hidden={true} dataField="id" isKey={true}>
          ID
        </TableHeaderColumn>

        <TableHeaderColumn width="200" dataField="name" dataSort={true}>
          Naam
          <br/>
          chauffeur &#x2195;
        </TableHeaderColumn>

        <TableHeaderColumn width="100" dataField="totalRides" dataSort={true}>
          Aantal
          <br/>
          ritten &#x2195;
        </TableHeaderColumn>

        <TableHeaderColumn width="130" dataField="typeOfPayment" dataSort={true}>
          Soort
          <br/>
          vergoeding &#x2195;
        </TableHeaderColumn>

        <TableHeaderColumn width="130" dataField="totalEarned" dataSort={true}>
          Totaal
          <br/>
          verdiend &#x2195;
        </TableHeaderColumn>

        <TableHeaderColumn width="130" dataField="numberPlate" dataSort={true}>
          Kenteken &#x2195;
        </TableHeaderColumn>

        <TableHeaderColumn width="120" dataField="numberOfPassengers" dataSort={true}>
          Aantal
          <br/>
          personen &#x2195;
        </TableHeaderColumn>

        <TableHeaderColumn width="130" dataField="segment" dataSort={true}>
          Segment
          <br/>
          auto &#x2195;
        </TableHeaderColumn>

        <TableHeaderColumn width="130" dataField="rating" dataSort={true}>
          Beoordeling &#x2195;
        </TableHeaderColumn>

      </BootstrapTable>

      <CSVLink data={this.state.data} filename={"chauffeurs_overview"} className="btn btn-primary crud-btn" target="">
        Export als CSV
      </CSVLink>
    </div>);
  };
}
