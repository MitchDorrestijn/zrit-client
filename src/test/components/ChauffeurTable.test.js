import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import ChauffeurTable from '../../main/components/ChauffeurTable';

const wrapper = shallow(<ChauffeurTable history="" routes="" />);

const push = jest.fn();
wrapper.setProps({ history: { push } });

test('CRUDTable component should match snapshot', () => {
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test('Should render nine TableHeaderColumn components', () => {
  expect(wrapper.find('TableHeaderColumn').length).toBe(9);
});

test('Should render one BootstrapTable component', () => {
  expect(wrapper.find('BootstrapTable').length).toBe(1);
});

test('Should render one CSVLink component', () => {
  expect(wrapper.find('CSVLink').length).toBe(1);
});
