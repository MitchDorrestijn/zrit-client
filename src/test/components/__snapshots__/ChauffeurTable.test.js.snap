// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CRUDTable component should match snapshot 1`] = `
<div>
  <BootstrapTable
    autoCollapse={
      Object {
        "filter": false,
        "search": false,
        "sort": false,
      }
    }
    bodyContainerClass={null}
    bordered={true}
    cellEdit={
      Object {
        "afterSaveCell": undefined,
        "beforeSaveCell": undefined,
        "blurToEscape": false,
        "blurToSave": false,
        "mode": "none",
        "nonEditableRows": undefined,
      }
    }
    columnFilter={false}
    condensed={false}
    containerClass={null}
    csvFileName="spreadsheet.csv"
    data={Array []}
    deleteRow={false}
    expandColumnOptions={
      Object {
        "expandColumnBeforeSelectColumn": true,
        "expandColumnComponent": undefined,
        "expandColumnVisible": false,
        "expandedColumnHeaderComponent": undefined,
      }
    }
    exportCSV={false}
    fetchInfo={
      Object {
        "dataTotalSize": 0,
      }
    }
    footer={false}
    headerContainerClass={null}
    height="100%"
    hover={false}
    ignoreSinglePage={false}
    insertRow={false}
    keyBoardNav={false}
    multiColumnSearch={false}
    multiColumnSort={1}
    options={
      Object {
        "btnGroup": [Function],
        "noDataText": "Geen resultaten gevonden",
        "renderSortedColumn": [Function],
        "searchField": [Function],
        "sortColumnName": undefined,
        "sortOrder": undefined,
      }
    }
    pagination={false}
    printable={false}
    renderAlert={true}
    replace={false}
    search={true}
    selectRow={
      Object {
        "mode": "radio",
        "onSelect": undefined,
      }
    }
    striped={false}
    tableBodyClass={null}
    tableContainerClass={null}
    tableFooterClass={null}
    tableHeaderClass={null}
    trClassName=""
    version="3"
    withoutTabIndex={false}
  >
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      csvFieldType="string"
      dataAlign="left"
      dataField="id"
      dataSort={false}
      defaultASC={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={true}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={true}
      keyValidator={false}
      searchable={true}
      sortIndicator={true}
      width={null}
    >
      ID
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      csvFieldType="string"
      dataAlign="left"
      dataField="name"
      dataSort={true}
      defaultASC={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      searchable={true}
      sortIndicator={true}
      width="200"
    >
      Naam
      <br />
      chauffeur ↕
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      csvFieldType="string"
      dataAlign="left"
      dataField="totalRides"
      dataSort={true}
      defaultASC={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      searchable={true}
      sortIndicator={true}
      width="100"
    >
      Aantal
      <br />
      ritten ↕
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      csvFieldType="string"
      dataAlign="left"
      dataField="typeOfPayment"
      dataSort={true}
      defaultASC={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      searchable={true}
      sortIndicator={true}
      width="130"
    >
      Soort
      <br />
      vergoeding ↕
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      csvFieldType="string"
      dataAlign="left"
      dataField="totalEarned"
      dataSort={true}
      defaultASC={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      searchable={true}
      sortIndicator={true}
      width="130"
    >
      Totaal
      <br />
      verdiend ↕
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      csvFieldType="string"
      dataAlign="left"
      dataField="numberPlate"
      dataSort={true}
      defaultASC={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      searchable={true}
      sortIndicator={true}
      width="130"
    >
      Kenteken ↕
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      csvFieldType="string"
      dataAlign="left"
      dataField="numberOfPassengers"
      dataSort={true}
      defaultASC={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      searchable={true}
      sortIndicator={true}
      width="120"
    >
      Aantal
      <br />
      personen ↕
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      csvFieldType="string"
      dataAlign="left"
      dataField="segment"
      dataSort={true}
      defaultASC={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      searchable={true}
      sortIndicator={true}
      width="130"
    >
      Segment
      <br />
      auto ↕
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      csvFieldType="string"
      dataAlign="left"
      dataField="rating"
      dataSort={true}
      defaultASC={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      searchable={true}
      sortIndicator={true}
      width="130"
    >
      Beoordeling ↕
    </TableHeaderColumn>
  </BootstrapTable>
  <CSVLink
    className="btn btn-primary crud-btn"
    data={Array []}
    filename="chauffeurs_overview"
    separator=","
    target=""
    uFEFF={true}
  >
    Export als CSV
  </CSVLink>
</div>
`;
